docs/development/extending.md

```diff
@@ -355,3 +355,486 @@ sorted_ip_list: "{{ ip_list | computacenter.utils.sort_ip }}"
+    # Function sorts a given list of IP addresses
+    
+        if not isinstance(unsorted_ip_list, list): # (5)!
+            raise AnsibleError("Filter needs list input, got '%s'" % type(unsorted_ip_list))
+        else:
+            sorted_ip_list = sorted(unsorted_ip_list, key=netaddr.IPAddress) # (6)!
+            
+        return sorted_ip_list # (7)!
+
+
+    class FilterModule(object): # (8)!
+        
+        def filters(self):
+            return {
+                # Sorting list of IP Addresses
+                'sort_ip': sort_ip # (9)!
+            }
+    ```
+
+    1. This is the most generic [AnsibleError object](https://github.com/ansible/ansible/blob/devel/lib/ansible/errors/__init__.py){:target="_blank"}, depending on the specific plugin type you’re developing you may want to use different ones.
+    2. Use this to convert plugin output to convert output into Python’s unicode type (*to_text*) or for wrapping other exceptions into error messages (*to_native*).
+    3. This is a non-standard dependency, the user needs to install this beforehand (e.g. `pip3 install netaddr --user`), therefor surrounding it with *try-execpt*. **Document necessary requirements!**
+    4. Example plugin definition, this sorts a given list of IP addresses ( Jinja2 *sort* filter does not work correctly with IPs), it expects a list.
+    5. Testing if input is a *list*, otherwise return an error message. Maybe another error type (e.g. *AnsibleFilterTypeError*) is more approriate? What other exceptions need to be caught?
+    6. This line sorts the list with the [*built-in Python sorted()*](https://docs.python.org/3/library/functions.html#sorted){:target="_blank"} library, the key specifies the comparison key for each list element, it uses the *netaddr* library.
+    7. The function returns a sorted list of IPs.
+    8. Main class, this is called by Ansible's *PluginLoader*.
+    9. Mapping of filter name and definition, you may call your filter like this: `"{{ ip_list | sort_ip }}"` (this only works when stored in the project root in the folder `filter_plugins`, otherwise you need to use the FQCN!). Filter name and definition do not need to have the same name.  Add more filter definitions by comma-separation.
+
+The Python file needs to be stored in a collection, e.g.:
+
+```bash
+collections/
+└── ansible_collections
+    └── computacenter
+        └── utils
+            ├── README.md
+            ├── docs
+            ├── galaxy.yml
+            ├── plugins
+            │   ├── README.md
+            │   └── filter
+            │       └── cc_filter_plugins.py
+            └── roles
+```
+
+!!! tip
+    Only underscores (`_`) are allowed for filenames inside collections!  
+    Naming the file `cc-filter-plugins.py` will result in an error!
+
+Now, the filter can be used:
+
+```yaml
+sorted_ip_list: "{{ ip_list | computacenter.utils.sort_ip }}"
+```
+
+!!! tip
+    The usage of the FQCN for your Plugin is mandatory!
+
+### Dynamic inventory plugins
```

Comment:
Der Punkt dynamisches Inventory würde ich auch eher im Punkt [Ansible > Inventory](https://timgrt.github.io/Ansible-Best-Practices/ansible/inventory/) sehen, dort einen neuen Unterpunkt erstellen.

Reply:
Ich würde es in der Datei extending lassen, da es sich hier um eine Entwicklung von einem dynamischen inventory handelt. Vielleicht machen wir eher eine Verlinkung im Inventory Absatz um auf die Entwicklung aufmerksam zu machen.

Reply:
Ok, einverstanden, habe eine entsprechende Referenz in einem Commit angefügt.